### ISSUE TITLE
Node count unexpectedly increases when create or edit GKE cluster

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#18069]
Version: Master 2.2 (2/13/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create a GKE cluster with defaults (3 nodes) and with node pool autoscaling 1 min 3 max.

**Result:**
You should now have 6 nodes according to Google Cloud admin console. This will continue as you edit more. 

**Other details that may be helpful:**
The front end looks correct we are sending a PUT to the server with the correct information in googleKubernetesEngineConfig as far as I can tell.

You might need to edit the cluster to see this (change something small)
Might not always occur especially if cluster is in error state. For example I had mis-configured my network and want to switch back to default but this does not fix the issue. Perhaps being in an error state has something to do with this as well.

![image](https://user-images.githubusercontent.com/45179589/52739183-96e51300-2f8d-11e9-87c6-0196bd37fa51.png)

![image](https://user-images.githubusercontent.com/45179589/52739195-9e0c2100-2f8d-11e9-887b-50dcaf1e13b2.png)

Each of these should be 3 nodes only.

**Comment by davidnuzik** [comment]
Possibly related to #15921 not sure

**Comment by vincent99** [comment]
Does it stay that way?  They could be deploying 3 new machines and eventually removing the 3 old ones.

**Comment by davidnuzik** [comment]
But if I asked for nodeCount of 3 and min 1 and max 3 I should still only get 3 upon cluster creation, no? It doesn't make sense from an end-user perspective to me that I should ever get any more than 3. It might go down to 3 after a while. This doesn't explain why one I had had 9 nodes.

**Comment by vincent99** [comment]
You're talking about editing though, a lot of edits are going to require making new nodes, and an obvious way to do that is make a new pool and then get rid of the old one. (I have no idea if that actually is what's happening, just speculating)

**Comment by davidnuzik** [comment]
That makes sense now. I do think something else is going on though, I just created one without any edit and had 4 nodes instead of 3 with same options (nodeCount 3 min 1 max 3). Didn't spot any issues making these nodes (such as if it had to tear one down and make a new one).

**Comment by jiaqiluo** [comment]
The bug exists on Rancher: master 2/15
Step:
- add a GKE cluster with `node count = 2`, `Minimum Node Count = 1` and `Maximum Node Count = 2`

<img width="1466" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/52877024-156cbc80-3116-11e9-9aa7-9b3a9543ec65.png">

Result:
- I get a GKE cluster with 4 nodes.
 
<img width="1358" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/52877036-1f8ebb00-3116-11e9-8d72-dad1c999dfc6.png">
<img width="534" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/52877060-2c131380-3116-11e9-99a1-344ad1484ef9.png">

**Comment by fyery-chen** [comment]
OK, I will check it immediately.

**Comment by fyery-chen** [comment]
@davidnuzik @jiaqiluo I cannot reproduce it, BTW, please ensure that you didn't set additional zone, because if you've added an additional zone then the number of nodes would increase by a corresponding multiple.

**Comment by loganhz** [comment]
Please see https://github.com/rancher/rancher/issues/18069#issuecomment-464259600

Added the to-test label so QA can notice it

**Comment by davidnuzik** [comment]
Thanks everyone, so when we ask for additional zones GKE (not rancher) is basically taking our nodeCount min/max and multiplying? Rancher is NOT doing this right? This is purely Google that is doing this.

When I edit my GKE cluster to reduce to only one zone (no additional zones) I still have 12 nodes. << That seems like a problem.

**Comment by fyery-chen** [comment]
Actually, our backend doesn't support update the additional zone, so about this problem, I want to refactor the update process according to the [GKE update API](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.zones.clusters/update), but I didn't know why not use this API before. Therefore I will submit a new issue about the refactoring of the GKE update API.

**Comment by davidnuzik** [comment]
Version: Master 2.2 (2/16/19)
Confirmed issue. This is happening because we are selecting more than one zone via the new "Additional Zones" checkbox options when creating a GKE cluster. I tested by creating several clusters some without additional zones and some with. Only when I ever selected one or more additional zones did I ever get more than the 3 expected nodes.

A new issue is going to be entered as well per @fyery-chen 's comment above. When you edit a Google GKE cluster to remove those additional zones the nodes stay behind and still exist.
